dvc/command/config.py

```diff
@@ -67,6 +67,13 @@ def _list(self):
             return 1
 
         levels = [self.args.level] if self.args.level else self.config.LEVELS
+        if self.config.dvc_dir is None:
+            levels = [
+                level
+                for level in levels
+                if level not in self.config.REPO_ONLY_LEVELS
+            ]
```

Comment:
I think it's a bit easier to read if we just set `levels` to what we want from the start, instead of setting it and then filtering it afterwards
```suggestion
        if self.args.level:
            levels = [self.args.level]
        elif self.config.dvc_dir is None:
            levels = list(self.config.REPO_ONLY_LEVELS)
        else:
            levels = list(self.config.LEVELS)
```

Alternatively, we could keep the list comprehension usage with something like

```py
@property 
def levels(self):
    return self.config.LEVELS if self.dvc_dir is not None else self.config.REPO_ONLY_LEVELS

...

def _list():
    ...
    levels = [self.args.level] if self.args.level else self.levels
```

---

Similar changes should be done in `_get` (but keep in mind the lists need to be reversed for `_get`)